### PR TITLE
add e2e private pools sample

### PR DESCRIPTION
**Pull request by agmsb**
Hello! Victory Szalvay and Damith Karunaratne asked me to open this PR to add a private pools example. Please let me know if I need to alter anything here!

### CODE REVIEW

File: private-pools-example/README.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
+# Accessing private resources using Cloud Build Private Pools
+
+In this repository, we will demonstrate the below functionality of Cloud Build Private Pools:
```
Comment (nityaravi) [inline private-pools-example/README.md:3]: This sample demonstrates the following features of Cloud Build private pools:

File: private-pools-example/README.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+* Private connectivity to resources in your VPC
```
Comment (nityaravi) [inline private-pools-example/README.md:5]: VPC network

File: private-pools-example/README.md (hunk 1 above)
Comment (nityaravi) [inline private-pools-example/README.md:3]: link to: https://cloud.google.com/build/docs/private-pools/private-pools-overview

Comment (agmsb) [reply private-pools-example/README.md]: Updated.

File: private-pools-example/README.md (hunk 2 above)
Comment (agmsb) [reply private-pools-example/README.md]: Updated.

File: private-pools-example/README.md (hunk 1 above)
Comment (agmsb) [reply private-pools-example/README.md]: Updated.

File: private-pools-example/README.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+In line 2 of `privatepool.yaml`, replace `<add-project-number>` and `<add-vpc-name>` with your GCP Project number and the value stored in `$VPC_NAME` respectively. 
+
+```
+    peeredNetwork: projects/<add-project-number>/global/networks/<add-vpc-name>
+```
+
+If you prefer to change the machine type, disk size, or disable Public IPs on the Cloud Build Private Pool, you can do so in lines 4-6. 
+
+```
+gcloud alpha builds worker-pools create $PRIVATE_POOL_NAME --config-from-file privatepool.yaml --region $REGION
```
Comment (nityaravi) [inline private-pools-example/README.md]: use the GA version of this command: https://cloud.google.com/build/docs/private-pools/create-manage-private-pools

File: private-pools-example/cloudbuild.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+steps:
+- name: 'gcr.io/cloud-builders/curl'
+  args: ['<add-ip-address>:80']
+
+# run build on private pool
+options:
```
Comment (nityaravi) [inline private-pools-example/cloudbuild.yaml:20]: use the GA verison of the config: https://cloud.google.com/build/docs/private-pools/run-builds-in-private-pool#running_builds

File: private-pools-example/README.md (hunk 3 above)
Comment (agmsb) [reply private-pools-example/README.md]: Up to date.

Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/97d3ee671c16fbf95a5ddbce21d0b41a466cf2ab

File: private-pools-example/cloudbuild.yaml (hunk 4 above)
Comment (agmsb) [reply private-pools-example/cloudbuild.yaml:20]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/97d3ee671c16fbf95a5ddbce21d0b41a466cf2ab

File: private-pools-example/README.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+
+This walkthrough requires that you have the below prerequisites:
+
+* Unix workstation
+* gcloud
+* kubectl
+* GCP Project with Cloud Build API, Service Networking API, and Kubernetes Engine API enabled
+* Permissions to create VPC networks, GKE clusters, and run builds in Cloud Build
+
+It is highly recommended you utilize a sandbox/test environment for this walkthrough. Cloud Shell is a suitable workstation for this tutorial.
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+
+## Prerequisites
+
+This walkthrough requires that you have the below prerequisites:
+
+* Unix workstation
+* gcloud
+* kubectl
+* GCP Project with Cloud Build API, Service Networking API, and Kubernetes Engine API enabled
+* Permissions to create VPC networks, GKE clusters, and run builds in Cloud Build
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+
+* Running a private sample application on a GKE cluster, exposed on a private RFC1918 IP address via an Internal Load Balancer
+* Accessing the sample application from a build running on Cloud Build Private Pools
+
+## Prerequisites
+
+This walkthrough requires that you have the below prerequisites:
+
+* Unix workstation
+* gcloud
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+
+In this repository, we will demonstrate the below functionality of Cloud Build Private Pools:
+
+* Private connectivity to resources in your VPC
+* Running builds within a specific region on Google-managed workers
+* Large range of machine types configurable for workers that run your build
+
+![Cloud Build Private Pools Architecture](img/architecture.png)
+
+The above functionality in Private Pools is achieved with the same schema users are used to in Cloud Build.
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 9]
```diff
@@ -0,0 +1,287 @@
+# Accessing private resources using Cloud Build Private Pools
+
+In this repository, we will demonstrate the below functionality of Cloud Build Private Pools:
+
+* Private connectivity to resources in your VPC
+* Running builds within a specific region on Google-managed workers
+* Large range of machine types configurable for workers that run your build
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 10]
```diff
@@ -0,0 +1,287 @@
+# Accessing private resources using Cloud Build Private Pools
+
+In this repository, we will demonstrate the below functionality of Cloud Build Private Pools:
+
+* Private connectivity to resources in your VPC
+* Running builds within a specific region on Google-managed workers
```
[the remark this answers is not in the document]
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/e657c3acb623fdfcd6fecf70f19b5f4b2fbab823

File: private-pools-example/README.md [hunk 11]
```diff
@@ -0,0 +1,294 @@
...
+Let's now create the Private Pool. 
+
+```
+gcloud builds worker-pools create $PRIVATE_POOL_NAME --config-from-file privatepool.yaml --region $REGION
+```
+
+Your output should look similar to the below.
+```
+Creating worker pool...done.                                                                                                              
+Created [https://cloudbuild.googleapis.com/v1beta1/projects/agmsb-k8s/locations/us-west1/workerPools/projects%2F381015962062%2Flocations%2Fus-west1%2FworkerPools%2Fsandbox-privatepool].
```
Comment (dkarunaratne) [inline private-pools-example/README.md:209]: There's a beta reference here.

File: private-pools-example/README.md [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+## Cleanup
+
+Delete the GKE cluster.
+```
+gcloud container clusters delete $GKE_CLUSTER_NAME --region $REGION
+```
+
+Delete the Cloud Build Private Pool. 
+```
+gcloud alpha builds worker-pools delete $PRIVATE_POOL_NAME --region $REGION
```
Comment (dkarunaratne) [inline private-pools-example/README.md:287]: Can we remove this alpha reference?

Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/5b8ea2b1e4beacca4edf5eea39f119d396016fcf

File: private-pools-example/README.md (hunk 11 above)
Comment (agmsb) [reply private-pools-example/README.md]: Updated in https://github.com/GoogleCloudPlatform/cloud-build-samples/pull/13/commits/5b8ea2b1e4beacca4edf5eea39f119d396016fcf

**Comment by dkarunaratne** [comment]
LGTM

**Comment by agmsb** [comment]
@nityaravi please let me know if there are any outstanding changes you'd like to see!

**Comment by nityaravi** [comment]
> @nityaravi please let me know if there are any outstanding changes you'd like to see!

Thanks for addressing the comments! We'll also need a DPE to review the code sample, and our DPE owner (dinagraves) is ooo. She'll be back tomorrow - can this wait until then?

**Comment by vszal** [comment]
> > @nityaravi please let me know if there are any outstanding changes you'd like to see!
> 
> Thanks for addressing the comments! We'll also need a DPE to review the code sample, and our DPE owner (dinagraves) is ooo. She'll be back tomorrow - can this wait until then?

Tomorrow is great. Thanks for your efforts @nityaravi

**Comment by vszal** [comment]
@dinagraves would you be able to give this a review?

**Comment by dinagraves** [comment]
Hi Victor! Thank you for this.  I do wonder if this would be better served as a tutorial rather than a straight sample?  The README is very long and would be better if it just pointed to a tutorial in the docs.  

Ideally, the sample should be testable as well.  In this case, rather than instruct the user to put the IP address in the cloudbuild, we'd add a step that automatically extracted and supplied the IP address (example [here](https://docs.google.com/document/d/17TSxlgf_sfselj_B3Uf9fOAq4O-zMTfjA2w4-ZHE-lo/edit?resourcekey=0-Z-J0ZyPVsFYKe2B86dLyIA#heading=h.15bsf6ts4sdr)) or we'd use substitutions to be able to pass in the IP address.    The first approach is preferable since substitutions are their own difficult subject!

**Comment by vszal** [comment]
@dinagraves I'll check in with @nityaravi on your docs tutorial suggestion and we'll get back to you.

**Comment by agmsb** [comment]
Checking in on this @dinagraves - should I just make this a community tutorial instead?

**Comment by agmsb** [comment]
Okay! I'll close this out for now @dinagraves and open a PR against /community.

**Pull request closed without merging**